### ISSUE TITLE
CRW only works on Little endian platforms

### ISSUE THREAD

**Issue by klauspost** [klauspost/rawspeed#83]
There are casts from *byte to *int in CiffEntry::getIntArray and similar, and they will not work correctly on big endian platforms.

**Comment by pedrocr** [comment]
Yes, all of CRW is little endian only. My plan was to change both the CIFF and TIFF parsers to use things like the get4LE macros so that they are endian-clean without special cases (the BE classes). darktable only runs on x86 though so I'd need to get a minimal test case setup to be able to test this on other architectures. 

What do you think about having a minimal dcraw style program that outputs ppm or something like it inside rawspeed itself? It would make it easier to have minimal test cases for this kind of stuff.

**Comment by klauspost** [comment]
The reason I don't use get4LE is for speed. 

Right now, if endianness matches the platform (as CRW), returning an array is simply returning a pointer to memory, instead of having to go through the entire array.

_However_, I don't think it gives a big advantage, so I am still undecided on whether it should just be dropped for a branching version.

**Comment by pedrocr** [comment]
The annoying places are all the others (e.g., getting the number of entries in the IFD) and for those get4LE type macros can be optimized out when the architecture matches so there's nothing lost. 

That just leaves the getXArray() functions and we can just do what we do now but in a single function instead of one in TiffEntry and another in TiffEntryBE. We could even do the swap in place instead of what we do now and save on the memory allocation.
